be/src/storage/lake/compaction_task.h

```diff
@@ -39,6 +40,8 @@ class CompactionTask {
 
     virtual Status execute(CancelFunc cancel_func, ThreadPool* flush_pool = nullptr) = 0;
 
+    Status execute_index_major_compaction(std::shared_ptr<TxnLogPB>& txn_log);
```

Comment:
is it necessary to use reference for shared_ptr? if don't want to track the object lifecycle, use raw pointer, if want to retain the lifecycle, use `shared_ptr`, not `shared_ptr &`

It seems that the method will retain the reference of the object, so the caller will not worry about the object's lifecycle management, but actually it doesn't.